gymnasium/vector/vector_env.py

```diff
@@ -235,81 +235,197 @@ class VectorWrapper(VectorEnv):
+    def action_space(
+        self,
+    ) -> spaces.Space[VectorActType] | spaces.Space[VectorWrapperActType]:
+        """Return the :attr:`Env` :attr:`action_space` unless overwritten then the wrapper :attr:`action_space` is used."""
+        if self._action_space is None:
+            return self.env.action_space
+        return self._action_space
+
+    @action_space.setter
+    def action_space(self, space: spaces.Space[VectorWrapperActType]):
+        self._action_space = space
+
+    @property
+    def observation_space(
+        self,
+    ) -> spaces.Space[VectorObsType] | spaces.Space[VectorWrapperObsType]:
+        """Return the :attr:`Env` :attr:`observation_space` unless overwritten then the wrapper :attr:`observation_space` is used."""
+        if self._observation_space is None:
+            return self.env.observation_space
+        return self._observation_space
+
+    @observation_space.setter
+    def observation_space(self, space: spaces.Space[VectorWrapperActType]):
+        self._observation_space = space
+
+    @property
+    def metadata(self) -> dict[str, Any]:
+        """Returns the :attr:`Env` :attr:`metadata`."""
+        if self._metadata is None:
+            return self.env.metadata
+        return self._metadata
+
+    @metadata.setter
+    def metadata(self, value: dict[str, Any]):
+        self._metadata = value
+
+    @property
+    def render_mode(self) -> str | None:
+        """Returns the :attr:`Env` :attr:`render_mode`."""
+        return self.env.render_mode
+
+    @property
+    def np_random(self) -> np.random.Generator:
+        """Returns the :attr:`Env` :attr:`np_random` attribute."""
+        return self.env.np_random
+
+    @np_random.setter
+    def np_random(self, value: np.random.Generator):
+        self.env.np_random = value
+
+    @property
+    def _np_random(self):
+        raise AttributeError(
+            "Can't access `_np_random` of a wrapper, use `.unwrapped._np_random` or `.np_random`."
+        )
+
+    @property
+    def unwrapped(self) -> VectorEnv[VectorObsType, VectorActType, VectorArrayType]:
+        """Returns the base :class:`VectorEnv`."""
         return self.env.unwrapped
 
     def __repr__(self):
+        """Returns a representation of the vector wrapper."""
         return f"<{self.__class__.__name__}, {self.env}>"
 
     def __del__(self):
+        """Calls :meth:`__del__` on the :attr:`env`."""
         self.env.__del__()
 
 
-class VectorObservationWrapper(VectorWrapper):
+class VectorObservationWrapper(
+    VectorWrapper[VectorWrapperObsType, VectorActType, VectorArrayType]
+):
     """Wraps the vectorized environment to allow a modular transformation of the observation. Equivalent to :class:`gym.ObservationWrapper` for vectorized environments."""
 
-    def reset(self, **kwargs):
-        observation = self.env.reset(**kwargs)
-        return self.observation(observation)
+    def __init__(self, env: VectorEnv[VectorObsType, VectorActType, VectorArrayType]):
```

Comment:
Does adding this actually change anything?

Reply:
(I'm like 90% sure it doesn't, and in that case I think it's better to just omit it)